src/gui/settings/qgssettingseditorwidgetwrapper.h

```diff
@@ -80,6 +82,19 @@ class GUI_EXPORT QgsSettingsEditorWidgetWrapper : public QObject
      */
     virtual void setWidgetFromVariant( const QVariant &value ) const = 0;
 
+    /**
+     * Enables automatic update, which causes the setting to be updated immediately when the widget
+     * value is changed.
+     *
+     * If a \a dialog is provided, the setting will be updated when the dialog is accpeted.
+     * If not, the setting will be updated directly at each widget value change.
+     *
+     * \note This must called after createEditor() or configureEditor().
+     *
+     * \since QGIS 3.40
+     */
+    void enableAutomaticUpdate( QDialog *dialog = nullptr );
```

Comment:
I think we can find a better name for this function, maybe configureAutomaticUpdate or configureAutomaticUpdateBehavior.